pkg/controller/identity/iampolicy/controller.go

```diff
@@ -99,7 +104,14 @@ func (e *external) Observe(ctx context.Context, mgd resource.Managed) (managed.E
 	}
 
 	if meta.GetExternalName(cr) == "" {
-		return managed.ExternalObservation{}, nil
+		// If external name not set there is still a change it may already exist
+		// Try to get the policy by name
+		policyArn, policyErr := e.getPolicyArnByNameAndPath(ctx, cr.Spec.ForProvider.Name, cr.Spec.ForProvider.Path)
+		if policyArn == nil || policyErr != nil {
+			return managed.ExternalObservation{}, awsclient.Wrap(policyErr, errExternalName)
+		}
+		meta.SetExternalName(cr, aws.ToString(policyArn))
```

Comment:
If it's possible to construct the whole ARN, maybe we should consider using external-name as the name of the policy and construct ARN wherever needed?

Reply:
Do we have to make it cheaper to get the account id to do this?

Reply:
What do you mean? Like storing account id in status to not query every time?

Reply:
In status, or the providerconfig(?). If we need to do an extra AWS API call for every observe.

Reply:
We could possibly store it in the status so that it's available everywhere and query only if that status field is empty.